src/reverse/BasicTypes.h

```diff
@@ -132,14 +132,14 @@ struct TweakDBID
 
     TweakDBID(const std::string_view aName)
     {
-        name_hash = RED4ext::CRC32(aName.data(), 0);
+        name_hash = RED4ext::CRC32(reinterpret_cast<const uint8_t*>(aName.data()), aName.size(), 0);
```

Comment:
I just had a better look now because of what was raised on Discord, and if this is done only because some characters in string may get interpreted as negative value and go out of the array somehow, think the fix should be in the RED4ext.SDK and not here

Reply:
Ok, there is an issue of no null terminator actually in some instances from discussion so this comment is invalid.